### PR TITLE
Sass 751

### PR DESCRIPTION
**Pull request by Dula-J**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you addressed warnings where appropriate?

##### After PRs been Reviewed
- [ ]  Have you checked the PR Builder passes?
- [ ]  Have you checked for merge conflicts?
- [ ]  Have you checked code coverage isn’t lower than previously?

### CODE REVIEW

**Comment by Dula-J** [comment]
DO NOT MERGE - Will need rebasing with SASS-753 once that pr is merged

File: app/common/SessionValues.scala [hunk 1]
```diff
@@ -23,6 +23,7 @@ object SessionValues {
   val DIVIDENDS_PRIOR_SUB = "DIVIDENDS_PRIOR_SUB"
   val INTEREST_PRIOR_SUB = "INTEREST_PRIOR_SUB"
   val GIFT_AID_PRIOR_SUB = "GIFT_AID_PRIOR_SUB"
+  val EMPLOYMENT_PRIOR_SUB = "EMPLOYMENT_PRIOR_SUB"
```
Comment (michael-rob-poole) [inline app/common/SessionValues.scala:26]: This can be deleted again

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,1012 @@
...
+  "as an individual in english" when {
+
+    ".show" should {
+
+      "return single employment summary view when there is only one employment without Expenses and Benefits" which {
+
+        lazy val playSessionCookies = PlaySessionCookieBaker.bakeSessionCookie(Map(
+          SessionValues.TAX_YEAR -> taxYear.toString,
+          SessionValues.EMPLOYMENT_PRIOR_SUB -> Json.prettyPrint(Json.toJson(singleEmploymentModel))
+        ))
```
Comment (michael-rob-poole) [inline it/controllers/employment/EmploymentSummaryControllerISpec.scala:49]: playSessionCookies Val can be deleted

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,1012 @@
...
+
+        lazy val result: WSResponse = {
```
Comment (michael-rob-poole) [inline it/controllers/employment/EmploymentSummaryControllerISpec.scala:51]: val result needs updating see SASS-753C for examples

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala (hunk 2 above)
Comment (Dula-J) [reply it/controllers/employment/EmploymentSummaryControllerISpec.scala:49]: Removed

File: app/common/SessionValues.scala (hunk 1 above)
Comment (Dula-J) [reply app/common/SessionValues.scala:26]: Removed

File: app/controllers/employment/EmploymentSummaryController.scala [hunk 4]
```diff
@@ -43,6 +42,7 @@ class EmploymentSummaryController @Inject()(
   def show(taxYear: Int) : Action[AnyContent] = authAction { implicit user =>
 
     val source: Option[AllEmploymentData] = getModelFromSession[AllEmploymentData](SessionValues.EMPLOYMENT_PRIOR_SUB)
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply app/controllers/employment/EmploymentSummaryController.scala:44]: Refactored now

File: it/utils/ViewHelpers.scala [hunk 5]
```diff
@@ -173,3 +172,131 @@ trait ViewHelpers { self: AnyWordSpec with Matchers =>
   }
 
 }
+
+object test extends App {
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply it/utils/ViewHelpers.scala:176]: Nope, removed

File: app/controllers/employment/EmploymentSummaryController.scala [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  implicit val executionContext: ExecutionContext = mcc.executionContext
+
+  def show(taxYear: Int) : Action[AnyContent] = authAction { implicit user =>
+
+    val source: Option[AllEmploymentData] = getModelFromSession[AllEmploymentData](SessionValues.EMPLOYMENT_PRIOR_SUB)
+
+    if(source.isDefined) {
+      val listOfEmployments = source.get.hmrcEmploymentData ++ source.get.customerEmploymentData
+      val doExpensesExist = if(source.get.hmrcExpenses.isDefined || source.get.customerExpenses.isDefined) true else false
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply app/controllers/employment/EmploymentSummaryController.scala:48]: Agreed

File: app/views/employment/MultipleEmploymentsSummaryView.scala.html [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    messages(s"employment.employmentSummary.heading")
+}
+
+@employmentRow(employmentName: String, changeLink: Call) = {
+<li class="hmrc-add-to-a-list__contents">
+    <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+        <a class="govuk-link" href="@changeLink">@employmentName</a>
+    </span>
+    <span class="hmrc-add-to-a-list__change">@messages("common.status.updated")</span>
+</li>
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply app/views/employment/MultipleEmploymentsSummaryView.scala.html:44]: We already have SummaryListRow in hmrc library but it doesn't seem to allow creating a row where the key field is an 'a' tag. I could create a 'General' template that could be shared here but i don't think its necessary here.

[the remark this answers is not in the document]
Comment (michael-rob-poole) [reply app/views/employment/MultipleEmploymentsSummaryView.scala.html:44]: ok - seems fine then

File: app/views/employment/MultipleEmploymentsSummaryView.scala.html [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+@expensesList() = {
+<div class="govuk-form-group">
+    <ul class="hmrc-add-to-a-list hmrc-add-to-a-list--short">
+        <li class="hmrc-add-to-a-list__contents">
+            <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                @if(expensesExist) {
+                    <a class="govuk-link" href="@controllers.employment.routes.CheckEmploymentExpensesController.show(taxYear)">
+                        @messages("employment.expenses.label")
+                    </a>
+                } else {<p>@messages("employment.expenses.label")</p>}
```
Comment (michael-rob-poole) [inline app/views/employment/MultipleEmploymentsSummaryView.scala.html:67]: <p> tag here creates extra margin in top and bottom

File: app/views/employment/SingleEmploymentSummaryView.scala.html [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                @if(employment.employmentData.isDefined){ @messages("common.status.updated") }else{ @messages("common.status.cannotUpdate")}
+            </span>
+        </li>
+        <li class="hmrc-add-to-a-list__contents">
+            <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                @employment.employmentBenefits.map{ _ =>
+                    <a class="govuk-link" href="@controllers.employment.routes.CheckYourBenefitsController.show(taxYear, employment.employmentId)">
+                    @messages("common.benefits")
+                    </a>
+                }.getOrElse{<p>@messages("common.benefits")</p>}
```
Comment (michael-rob-poole) [inline app/views/employment/SingleEmploymentSummaryView.scala.html:58]: <p> tag here creates extra margin in top and bottom

File: app/views/employment/SingleEmploymentSummaryView.scala.html [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+@employmentTable(employment: EmploymentSource) = {
+<div class="govuk-form-group">
+    <ul class="hmrc-add-to-a-list hmrc-add-to-a-list--short">
+        <li class="hmrc-add-to-a-list__contents">
+            <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                @employment.employmentData.map{ _ =>
+                    <a class="govuk-link" href="@controllers.employment.routes.EmploymentDetailsController.show(taxYear, employment.employmentId)">
+                    @messages("common.employmentDetails")
+                    </a>
+                }.getOrElse{<p>@messages("common.employmentDetails")</p>}
```
Comment (michael-rob-poole) [inline app/views/employment/SingleEmploymentSummaryView.scala.html:46]: <p> tag here creates extra margin in top and bottom

File: app/views/employment/SingleEmploymentSummaryView.scala.html [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+                @if(employment.employmentBenefits.isDefined){ @messages("common.status.updated") }else{ @messages("common.status.cannotUpdate")}
+            </span>
+        </li>
+        <li class="hmrc-add-to-a-list__contents">
+            <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                @if(expensesExist) {
+                    <a class="govuk-link" href="@controllers.employment.routes.CheckEmploymentExpensesController.show(taxYear)">
+                    @messages("employment.expenses.label")
+                    </a>
+                } else {<p>@messages("employment.expenses.label")</p>}
```
[the remark this answers is not in the document]
Comment (Dula-J) [reply app/views/employment/SingleEmploymentSummaryView.scala.html]: Good spot, removed

**Pull request merged**